Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -63,6 +63,40 @@ struct TextureMapperLayer::ComputeTransformData {
     }
 };
 
+class TextureMapperFlattenedLayer : public TextureMapperPlatformLayer {
+public:
+    TextureMapperFlattenedLayer(const IntRect& rect)
+        : m_rect(rect)
+    { }
+
+    void bind(TextureMapper& textureMapper)
+    {
+        if (!m_texture)
+            m_texture = textureMapper.acquireTextureFromPool(m_rect.size(), { BitmapTexture::Flags::SupportsAlpha });
+        textureMapper.bindSurface(m_texture.get());
+    }
+
+    inline FloatRect layerRect() const
+    {
+        return FloatRect(m_rect);
+    }
+
+    RefPtr<BitmapTexture> surface() const
```

Comment:
Why return a RefPtr<> here? That's unnecessary refcount churn, if you just want to access the surface and non-null is guaranteed, I'd suggest to expose 'BitmapTexture& surface() const { return *m_texture.get(); }'.

Reply:
Patch uses SetForScope helper to temporary switch current TextureMapperPaintOptions target surface which is type of RefPtr<>. SetForScope requires types to match